providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/ActivityLogAlertApiMockTest.java

```diff
@@ -0,0 +1,210 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jclouds.azurecompute.arm.domain.ActionGroup;
+import org.jclouds.azurecompute.arm.domain.Actions;
+import org.jclouds.azurecompute.arm.domain.ActivityLogAlert;
+import org.jclouds.azurecompute.arm.domain.ActivityLogAlertProperties;
+import org.jclouds.azurecompute.arm.domain.AlertRuleAllOfCondition;
+import org.jclouds.azurecompute.arm.domain.AlertRuleAnyOfOrLeafCondition;
+import org.jclouds.azurecompute.arm.domain.AlertRuleLeafCondition;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiMockTest;
+import org.testng.annotations.Test;
+
+import okhttp3.mockwebserver.MockResponse;
+
+@Test(groups = "unit", testName = "ActivityLogAlertApiMockTest", singleThreaded = true)
+public class ActivityLogAlertApiMockTest extends BaseAzureComputeApiMockTest {
+
+	public void testList() throws InterruptedException {
+		server.enqueue(jsonResponse("/activitylogalertresourcegroup.json"));
+		final ActivityLogAlertApi activityLogAlertApi = api.getActivityLogAlertApi("myResourceGroup");
+		assertEquals(activityLogAlertApi.list(), getActivityLogRuleList());
+		assertSent(server, "GET",
+				"/subscriptions/SUBSCRIPTIONID/resourcegroups/myResourceGroup/providers/Microsoft.Insights/activityLogAlerts?api-version=2020-10-01");
+	}
+
+	private List<ActivityLogAlert> getActivityLogRuleList() {
+		List<ActivityLogAlert> activityLogAlertRules = new ArrayList<ActivityLogAlert>();
+		activityLogAlertRules.add(ActivityLogAlert.create(
+				"/subscriptions/SUBSCRIPTIONID/resourceGroups/myResourceGroup/providers/microsoft.insights/activityLogAlerts/myActivityLog",
+				"myActivityLog", "Global", "Microsoft.Insights/ActivityLogAlerts", tags(),
+				myActivityLogAlertProperties()));
+		activityLogAlertRules.add(ActivityLogAlert.create(
+				"/subscriptions/SUBSCRIPTIONID/resourceGroups/myResourceGroup/providers/microsoft.insights/activityLogAlerts/simpleActivityLog",
+				"simpleActivityLog", "Global", "Microsoft.Insights/ActivityLogAlerts", null,
+				sampleActivityLogAlertProperties()));
+		return activityLogAlertRules;
+	}
+
+	public ActivityLogAlertProperties myActivityLogAlertProperties() {
+		return ActivityLogAlertProperties.create("", false, scope0(), condition0(), actions());
+	}
+
+	public ActivityLogAlertProperties sampleActivityLogAlertProperties() {
+		return ActivityLogAlertProperties.create("", true, scope1(), condition1(), actions());
+
+	}
+
+	public void testListEmpty() throws Exception {
+		server.enqueue(new MockResponse().setResponseCode(404));
+		final ActivityLogAlertApi activityApi = api.getActivityLogAlertApi("groupname");
+		List<ActivityLogAlert> list = activityApi.list();
+		assertTrue(isEmpty(list));
```

Comment:
Most of these `assertTrue` would be better written with `assertThat(list).isEmpty()` for example.

Reply:
I will use for future test cases.